src/EntryPoints/Specials/SpecialApproverCategories.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace ProfessionalWiki\PageApprovals\EntryPoints\Specials;
+
+use MediaWiki\MediaWikiServices;
+use ProfessionalWiki\PageApprovals\Adapters\DatabaseApproverRepository;
+use ProfessionalWiki\PageApprovals\Adapters\UsersLookup;
+use SpecialPage;
+use PermissionsError;
+use ProfessionalWiki\PageApprovals\Application\UseCases\GetAllApproversCategories;
+use ProfessionalWiki\PageApprovals\Application\UseCases\SetApproverCategories;
+use LightnCandy\LightnCandy;
+use WebRequest;
+
+class SpecialApproverCategories extends SpecialPage {
+
+	public function __construct() {
+		parent::__construct( 'ApproverCategories' );
+	}
+
+	public function isListed(): bool {
+		return $this->isAdmin();
+	}
+
+	public function execute( $subPage ): void {
+		if ( !$this->isAdmin() ) {
+			throw new PermissionsError( 'sysop' );
```

Comment:
If I'm right, PermissionsError takes the name of a permission/right, not the name of a user group. Sysop is a user group.

You can call `$this->checkPermissions()`, which will do the checks based on the permission/right you provided in the constructor, which would look something like

```php
parent::__construct( 'ApproverCategories', 'userrights' );
```

`userrights` here is an existing permission/right which we could plausibly use. You can see what permissions/rights are available in which user groups by default via http://localhost:8484/index.php/Special:ListGroupRights. Note that sysop is the internal name for Admionistrators

Reply:
We could later add our own permission/right, so wikis can be configured more granular, but that is definitely not needed in this PR

Reply:
You can see an example of how this check is done in the `AdminPanel`  class in our other codebase

Reply:
I created a separate ticket to take care of this since it is likely needed by many other PRs not just this one.

https://github.com/ProfessionalWiki/PageApprovals/issues/31

Reply:
I'm not sure where else you would need this. The issue in this PR is that you are adding a bunch of not needed code that reimplements the standard mechanism. You can replace it by 2 lines of code.

Reply:
If you don't understand I can push a commit on top of this branch when it's clear you are not touching it.

Reply:
If its only 2 lines of code. You can create PR and direct at this branch for handling 

`$this->checkPermissions();`